include/interface.h

```diff
@@ -0,0 +1,53 @@
+#ifndef MPM_INTERFACE_INTERFACE_H_
```

Comment:
Why do we need a class? can this not be saved directly if there is only one parameter?